app/src/components/market/market_wizard_creator_container.tsx

```diff
@@ -23,70 +27,81 @@ const MarketWizardCreatorContainer: FC = () => {
 
   const handleSubmit = async (data: MarketData) => {
     try {
-      if (!data.resolution) {
-        throw new Error('resolution time was not specified')
-      }
-
-      const networkId = context.networkId
-      const provider = context.library
-      const user = await provider.getSigner().getAddress()
-
-      const { collateral, arbitratorId, question, resolution, funding, outcomes, category } = data
-      const openingDateMoment = moment(resolution)
-
-      const arbitrator = getArbitrator(networkId, arbitratorId)
-
-      setStatus(StatusMarketCreation.PostingQuestion)
-      const questionId = await realitio.askQuestion(
-        question,
-        outcomes,
-        category,
-        arbitrator.address,
-        openingDateMoment,
-        networkId,
-      )
-      setQuestionId(questionId)
-
-      setStatus(StatusMarketCreation.PrepareCondition)
-
-      const oracleAddress = getContractAddress(networkId, 'oracle')
-      const conditionId = await conditionalTokens.prepareCondition(
-        questionId,
-        oracleAddress,
-        outcomes.length,
-      )
-
-      // approve movement of collateral token to MarketMakerFactory
-      setStatus(StatusMarketCreation.ApprovingCollateral)
-
-      const marketMakerFactoryAddress = getContractAddress(networkId, 'marketMakerFactory')
-      const collateralService = new ERC20Service(provider, collateral.address)
-
-      const hasEnoughAlowance = await collateralService.hasEnoughAllowance(
-        user,
-        marketMakerFactoryAddress,
-        funding,
-      )
-      if (!hasEnoughAlowance) {
-        await collateralService.approve(marketMakerFactoryAddress, funding)
+      if (!context.account) {
+        setModalState(true)
+      } else {
+        if (!data.resolution) {
+          throw new Error('resolution time was not specified')
+        }
+        const user = await provider.getSigner().getAddress()
+        const { collateral, arbitratorId, question, resolution, funding, outcomes, category } = data
+        const openingDateMoment = moment(resolution)
+
+        const collateralService = new ERC20Service(
+          provider,
+          rawWeb3Context.connectorName,
+          collateral.address,
+        )
+
+        const hasEnoughBalanceToFund = await collateralService.hasEnoughBalanceToFund(user, funding)
+        if (!hasEnoughBalanceToFund) {
+          throw new Error('there are not enough collateral balance for funding')
```

Comment:
Mmmm maybe we need to inform this error message to the user, What about to use the `<Message />` component ? Or maybe you are ok with the actual behaviour ?

Reply:
Done!